bezier-rs/lib/src/lib.rs

```diff
@@ -122,57 +135,86 @@ impl Bezier {
 			BezierHandles::Cubic { handle1, handle2 } => [Some(self.start), Some(handle1), Some(handle2), Some(self.end)],
 		}
 	}
 
-	/// Return the length of the bezier curve
-	pub fn length() -> i32 {
-		0
-	}
-}
+	///  Calculate the x-value of a point on the curve based on the t-value provided
+	///  basis code based off of pseudocode found here: https://pomax.github.io/bezierinfo/#explanation
+	pub fn get_x_basis(&self, t: f64) -> f64 {
+		let t2 = t * t;
+		let mt = 1.0 - t;
+		let mt2 = mt * mt;
 
-/*
+		match self.handles {
+			BezierHandles::Quadratic { handle } => mt2 * self.start[0] + 2.0 * mt * t * handle[0] + t2 * self.end[0],
+			BezierHandles::Cubic { handle1, handle2 } => {
+				let t3 = t2 * t;
+				let mt3 = mt2 * mt;
+				mt3 * self.start[0] + 3.0 * mt2 * t * handle1[0] + 3.0 * mt * t2 * handle2[0] + t3 * self.end[0]
+			}
+		}
+	}
 
-/// for computing the length of a bezier curve
-/// taken from https://pomax.github.io/bezierinfo/#arclength
+	///  Calculate the y-value of a point on the curve based on the t-value provided
+	///  basis code based off of pseudocode found here: https://pomax.github.io/bezierinfo/#explanation
+	pub fn get_y_basis(&self, t: f64) -> f64 {
+		let t2 = t * t;
+		let mt = 1.0 - t;
+		let mt2 = mt * mt;
 
-computeLength(curve) {
-	const z = 0.5, len = T.length;
-	let sum = 0;
-	for (let i = 0, t; i < len; i++) {
-	  t = z * T[i] + z;
-	  sum += C[i] * this.arcfn(t, curve.derivative(t));
+		match self.handles {
+			BezierHandles::Quadratic { handle } => mt2 * self.start[1] + 2.0 * mt * t * handle[1] + t2 * self.end[1],
+			BezierHandles::Cubic { handle1, handle2 } => {
+				let t3 = t2 * t;
+				let mt3 = mt2 * mt;
+				mt3 * self.start[1] + 3.0 * mt2 * t * handle1[1] + 3.0 * mt * t2 * handle2[1] + t3 * self.end[1]
+			}
+		}
 	}
-	return z * sum;
-}
 
-arcfn(t, d) {
-	return sqrt(d.x * d.x + d.y * d.y);
-}
+	/// Return an approximation of the length of the bezier curve
+	/// code example taken from: https://gamedev.stackexchange.com/questions/5373/moving-ships-between-two-planets-along-a-bezier-missing-some-equations-for-acce/5427#5427
+	pub fn length(&self) -> f64 {
+		// We will use an approximate approach where
+		// we split the curve into many subdivisions
+		// and calculate the euclidean distance between the two endpoints of the subdivision
+		const SUBDIVISIONS: i32 = 1000;
+		const RATIO: f64 = 1.0 / (SUBDIVISIONS as f64);
+
+		// ox, oy track the starting point of the subdivision
+		let mut ox = self.get_x_basis(0.0);
+		let mut oy = self.get_y_basis(0.0);
+		let mut clen = 0.0;
+		// calculate approximate distance between subdivision
+		for i in 1..SUBDIVISIONS + 1 {
+			// get end point of the subdivision
+			let x = self.get_x_basis(f64::from(i) * RATIO);
+			let y = self.get_y_basis(f64::from(i) * RATIO);
+			// calculate distance of subdivision
+			let dx = ox - x;
+			let dy = oy - y;
+			clen += (dx * dx + dy * dy).sqrt();
+			// update ox, oy for next subdivision
+			ox = x;
+			oy = y;
```

Comment:
Then this could be simplified to use points (e.g. `let mut old_positon: DVec2 = ...`) - you can also do things like `total_distance += (old_position - current_position).length();`